module_1/src/com/epam/cycles/CyclesTaskF.java

```diff
@@ -0,0 +1,44 @@
+package com.epam.cycles;
+import java.util.Scanner;
+
+public class CyclesTaskF {
+
+    public static void main(String[] args) {
+
+        Scanner console = new Scanner(System.in);
+        System.out.println("Set the interval [m,n] including natural numbers: ");
+        System.out.print("-- enter the value 'm': ");
+        int m = console.nextInt();
+        System.out.print("-- enter the value 'n': ");
+        int n = console.nextInt();
+
+        if (m < n && n >= 1) {
+
+            System.out.println("\nDivisors of numbers (excluding 1 and the same number):");
+            boolean checkNonNaturalNumber;
+
+            for (int currentNumber = m; currentNumber <= n; currentNumber++) {
+                if (currentNumber < 1) {
+                    continue;
+                }
+                checkNonNaturalNumber = false;
+                System.out.print("\nThe number " + currentNumber + " -> ");
+                int divisor = 2;
+                while (divisor < currentNumber) {
```

Comment:
You don't need to check till currentNumber, it is enough to check till currentNumber / 2
```suggestion
                while (divisor < currentNumber / 2) {
```